production.Dockerfile

```diff
@@ -46,70 +85,73 @@ RUN apk --update --no-cache add \
     && \
     apk del .build-deps
 
-# Copy everything else
-COPY . .
+RUN addgroup -S posthog && \
+    adduser -S posthog -G posthog
 
-# Compile and install Yarn dependencies.
-#
-# Notes:
-#
-# - we explicitly COPY the files so that we don't need to rebuild
-#   the container every time a dependency changes
-#
-# - we need few additional OS packages for this. Let's install
-#   and then uninstall them when the compilation is completed.
-COPY package.json yarn.lock ./
-RUN apk --update --no-cache --virtual .build-deps add \
-    "gcc~=10.3" \
-    && \
-    yarn config set network-timeout 300000 && \
-    yarn install --frozen-lockfile && \
-    yarn build && \
-    yarn cache clean && \
-    rm -rf ./node_modules &&\
-    yarn install --frozen-lockfile --cwd plugin-server && \
-    yarn cache clean && \
-    apk del .build-deps
+RUN chown posthog.posthog /code
 
+USER posthog
 
+# Add in Django deps and generate Django's static files
+COPY manage.py manage.py
+COPY posthog posthog/
+COPY ee ee/
+COPY --from=frontend /code/frontend/dist /code/frontend/dist
 
-# Build the plugin server
-#
-# Note: we run the build as a separate actions to increase
-# the cache hit ratio of the layers above.
-# symlink musl -> ld-linux is required for re2 compat on alpine
-RUN cd plugin-server \
-    && ln -s /lib/ld-musl-x86_64.so.1 /lib/ld-linux-x86-64.so.2 \
-    && yarn build \
+RUN SKIP_SERVICE_VERSION_REQUIREMENTS=1 SECRET_KEY='unsafe secret key for collectstatic only' DATABASE_URL='postgres:///' REDIS_URL='redis:///' python manage.py collectstatic --noinput
+
+# Add in the plugin-server compiled code, as well as the runtime dependencies
+WORKDIR /code/plugin-server
+COPY ./plugin-server/package.json ./plugin-server/yarn.lock ./
+
+# Switch to root and install yarn so we can install runtime deps. Node that we
+# still need yarn to run the plugin-server so we do not remove it.
+USER root
+RUN apk --update --no-cache add "yarn~=1"
+
+# NOTE: we need make and g++ for node-gyp
+# NOTE: npm is required for re2
+RUN apk --update --no-cache add "make~=4.3" "g++~=10.3" "npm~=7" --virtual .build-deps \
+    && yarn install --frozen-lockfile --production=true \
     && yarn cache clean \
-    && cd ..
+    && apk del .build-deps
 
-# Generate Django's static files
-RUN SKIP_SERVICE_VERSION_REQUIREMENTS=1 SECRET_KEY='unsafe secret key for collectstatic only' DATABASE_URL='postgres:///' REDIS_URL='redis:///' python manage.py collectstatic --noinput
+USER posthog
 
-# Add a dedicated 'posthog' user and group, move files into its home dir and set the
-# proper file permissions. This alleviates compliance issue for not running a
-# container as 'root'
-RUN addgroup -S posthog && \
-    adduser -S posthog -G posthog && \
-    mv /code /home/posthog && \
-    chown -R posthog:1000 /home/posthog/code
-WORKDIR /home/posthog/code
+# Add in the compiled plugin-server
+COPY --from=plugin-server /code/plugin-server/dist/ ./dist/
```

Comment:
Is it really the case that the plugin-server does not need `node_modules` ? I understand it for the frontend as that is all bundled code but surely nodejs uses it in the plugin server?

Reply:
I suspect it does need them, so we install them in the final image as well.

Note that we now also use the image in the E2E tests, which should catch this(?)

Reply:
( see https://github.com/PostHog/posthog/pull/10488/files#diff-55d06c59f3552514747959fc163ad2940d33f0757fa48cf363580af437dee6a3R131 )